src/library/tabledelegates/stareditor.cpp

```diff
@@ -83,31 +89,132 @@ void StarEditor::paintEvent(QPaintEvent*) {
         cg = QPalette::Inactive;
     }
 
+    painter.save();
     if (m_styleOption.state & QStyle::State_Selected) {
         painter.setBrush(m_styleOption.palette.color(cg, QPalette::HighlightedText));
     } else {
         painter.setBrush(m_styleOption.palette.color(cg, QPalette::Text));
     }
-
     m_starRating.paint(&painter, m_styleOption.rect);
-
-    // Draw a border if the color cell is selected
-    if (m_styleOption.state & QStyle::State_HasFocus) {
-        TableItemDelegate::drawBorder(&painter, m_focusBorderColor, m_styleOption.rect);
-    }
+    painter.restore();
 }
 
 bool StarEditor::eventFilter(QObject* obj, QEvent* event) {
     switch (event->type()) {
+    case QEvent::KeyPress: {
+        VERIFY_OR_DEBUG_ASSERT(m_isKeyboardEditMode) {
+            // Persistent editors (i.e. those opened using openPersistentEditor)
+            // should not receive keyboard events via their eventFilter, so we
+            // shouldn't normally arrive here - but if we do, just forward the events.
+            return false;
+        }
+
+        // Change rating when certain keys are pressed
+        // while we are in edit mode.
+        QKeyEvent* ke = static_cast<QKeyEvent*>(event);
+        int newRating = m_starRating.starCount();
+        switch (ke->key()) {
+        case Qt::Key_Up:
+        case Qt::Key_Down:
+        case Qt::Key_PageUp:
+        case Qt::Key_PageDown: {
+            // Allow handling of certain keyboard navigation events.
+            // This logic attempts to match the behavior for text editor cells.
+            return false;
+        }
+        case Qt::Key_Home:
+        case Qt::Key_End: {
+            // Only forward Home and End if the Ctrl key is pressed.
+            // This matches the behavior of normal text editor cells.
+            if (ke->modifiers() & Qt::ControlModifier) {
+                return false;
+            }
+            return true;
+        }
+        case Qt::Key_0: {
+            newRating = 0;
+            break;
+        }
+        case Qt::Key_1: {
+            newRating = 1;
+            break;
+        }
+        case Qt::Key_2: {
+            newRating = 2;
+            break;
+        }
+        case Qt::Key_3: {
+            newRating = 3;
+            break;
+        }
+        case Qt::Key_4: {
+            newRating = 4;
+            break;
+        }
+        case Qt::Key_5: {
+            newRating = 5;
+            break;
+        }
+        case Qt::Key_6: {
+            newRating = 6;
+            break;
+        }
+        case Qt::Key_7: {
+            newRating = 7;
+            break;
+        }
+        case Qt::Key_8: {
+            newRating = 8;
+            break;
+        }
+        case Qt::Key_9: {
+            newRating = 9;
+            break;
+        }
```

Comment:
(Nit: the curly bracket in your switch statement are redundant - you should only need them if you have nested control flow)